### PR TITLE
don't set readonly property of HTTPError

### LINKED ISSUE #1644: bug in httperror handling in dataset utils
in `nilearn.datasets.utils`, we try to set the `reason` attribute of an
`HTTPError` from `urllib`, which is a property without a setter. `HTTPErrors`
get transformed into `AttributeErrors`

```
    except (_urllib.error.HTTPError, _urllib.error.URLError) as e:
        if 'Error while fetching' not in str(e):
            # For some odd reason, the error message gets doubled up
            #   (possibly from the re-raise), so only add extra info
            #   if it's not already there.
            e.reason = ("%s| Error while fetching file %s; "
                          "dataset fetching aborted." % (
                            str(e.reason), file_name))
        raise
```

### PR DESCRIPTION
**Pull request by jeromedockes**
fixes #1644

### CODE REVIEW

**Comment by bthirion** [comment]
LGTM. No better suggestion on my side !

File: nilearn/datasets/utils.py [hunk 1]
```diff
@@ -575,15 +575,6 @@ def _fetch_file(url, data_dir, resume=True, overwrite=False,
...
                              .format(dt, dt // 60))
-    except (_urllib.error.HTTPError, _urllib.error.URLError) as e:
-        if 'Error while fetching' not in str(e):
-            # For some odd reason, the error message gets doubled up
-            #   (possibly from the re-raise), so only add extra info
-            #   if it's not already there.
-            e.reason = ("%s| Error while fetching file %s; "
-                          "dataset fetching aborted." % (
-                            str(e.reason), file_name))
-        raise
```
Comment (GaelVaroquaux) [inline nilearn/datasets/utils.py]: We should print this error message, which is more informative than the original HTTPError, and raise the original error, to enable error managment.

Comment (GaelVaroquaux) [reply nilearn/datasets/utils.py]: Print it with "sys.stderr.write()".

**Comment by jeromedockes** [comment]
close in favor of #1655 thanks @KamalakerDadi

**Pull request closed without merging**